src/fit-utils.jl

```diff
@@ -175,45 +175,40 @@ function split_set_threads!(
     lefts = zeros(Int, nblocks)
     rights = zeros(Int, nblocks)
 
-    @sync begin
-        for bid = 1:nblocks
-            @spawn begin
-                lefts[bid], rights[bid] = split_set_chunk!(
-                    left,
-                    right,
-                    is,
-                    bid,
-                    nblocks,
-                    x_bin,
-                    feat,
-                    cond_bin,
-                    feattype,
-                    offset,
-                    chunk_size,
-                )
-            end
-        end
+    @threads for bid = 1:nblocks
```

Comment:
This seems to go contrary to current advice: https://julialang.org/blog/2023/07/PSA-dont-use-threadid/ .

Reply:
I've read your comment below and stand corrected. Thanks for the education 😄 .